tutorials/cloud-iot-firestore-config/index.md

```diff
@@ -0,0 +1,301 @@
+1. For the field, type, and value use the following:
+
+![config-doc](./images/config-doc.png)
+
+Note that the different fields in the config can have different data types.  Save this document.
+
+Now open the [Cloud IoT Core console](https://console.cloud.google.com/iot/locations/us-central1/registries/config-demo/devices/sample-device), choose the device and look at the `Configuration & state history` pane:
+
+![config-choose](./images/choose-config.png)
+
+If the Function ran succesfully - you should be able to select, and see the initial configuration saved with the device.
+
+![initial config](./images/initial-config.png)
+
+## Modify the config
+
+Lets start up the sample device now in your shell
+
+```sh
+# still in the sample-device subfolder
+
+npm install
+
+node build/index.js
+```
+
+You should see output that looks like:
+
+```sh
+Device Started
+Current Config: 
+{ energySave: false, mode: 'heating', tempSetting: 35 }
+```
+
+Now lets update the config document in the Firestore console to change the tempSetting value to `18`
+
+When this document edit is saved, it triggers a function, which will push the new config down to the device:
+
+![field update](./images/update.png)
+
+You should see this new config arrive at the sample device in a second or two.
+
+```sh
+Current Config: 
+{ energySave: false, mode: 'heating', tempSetting: 18 }
+```
+
+To do this programatically with IoT Core APIs alone, you would have to read the current config from the IoT Core Device Manager, update the value, then write back the new config to IoT Core.  IoT Core does provide a incrementing version number you can send with these writes to check that another process has not concurrently attempted to update the config.
+
+This solution assumes that the path using Firestore and functions are not sharing the config update job with other processes, but are acting as a flexible intermediate.  Both single Firestore Documents and IoT Core device configurations are limited to one update per second.
+
+
+## Perform Queries with Firestore
+
+You can use the [query capabilities](https://firebase.google.com/docs/firestore/query-data/queries) of Firestore to find devices with specific configurations:
+
+```js
+var configs = db.collection('device-configs');
+var hotDevices = configs.where('tempSetting', '>', 40);
+```
+
+The above snippet is for nodejs, but see the [Firestore quickstart](https://firebase.google.com/docs/firestore/quickstart) for how to set up and query from a number of different runtimes.
+
+
+## Binary data with CBOR
+
+Sometimes on with constrained devices and constrained networks, you want to work with data in a compact binary format.  Concise Binary Object Representation [CBOR](http://cbor.io/) is a binary format that strikes a balance between the compactness of binary, with the self describing format of JSON. IoT Core device configuration API and MQTT both fully support binary messages.
```

Comment:
I would say "Sometimes, on constrained devices" or "Sometimes, with constrained devices" instead of having on with